From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Tue, 15 Mar 2022 17:49:22 -0400
Subject: [PATCH] Rename bladeRF.dll to bladeRF-2.dll on Windows.

This allows installing multiple major versions of the library on
Windows, which brings feature parity to the Unix builds for packaging
into a libbladerf2 package.
---
 host/libraries/libbladeRF/CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git host/libraries/libbladeRF/CMakeLists.txt host/libraries/libbladeRF/CMakeLists.txt
index dea3b3cf..a16bce5e 100644
--- host/libraries/libbladeRF/CMakeLists.txt
+++ host/libraries/libbladeRF/CMakeLists.txt
@@ -451,6 +451,10 @@ target_link_libraries(libbladerf_shared ${LIBBLADERF_LIBS})
 # Adjust our output name
 set_target_properties(libbladerf_shared PROPERTIES OUTPUT_NAME bladeRF)
 
+if(MSVC)
+    set_target_properties(libbladerf_shared PROPERTIES RUNTIME_OUTPUT_NAME "bladeRF-${VERSION_INFO_MAJOR}")
+endif()
+
 # Set shared library version
 set_target_properties(libbladerf_shared PROPERTIES SOVERSION ${VERSION_INFO_MAJOR})
 
@@ -477,6 +481,9 @@ set(LIBBLADERF_PC_LIBDIR \${exec_prefix}/${CMAKE_INSTALL_LIBDIR})
 set(LIBBLADERF_PC_INCLUDEDIR \${prefix}/include)
 set(LIBBLADERF_PC_VERSION ${VERSION})
 set(LIBBLADERF_PC_LIBS "-lbladeRF")
+if(MSVC)
+    set(LIBBLADERF_PC_LIBS "-lbladeRF-${VERSION_INFO_MAJOR}")
+endif()
 
 # Use space-delimiter in the .pc file, rather than CMake's semicolon separator
 string(REPLACE ";" " " LIBBLADERF_PC_CFLAGS "${LIBBLADERF_PC_CFLAGS}")
-- 
2.35.0

